src/class-drip-woocommerce-cart-events.php

```diff
@@ -43,10 +43,14 @@ public function drip_woocommerce_cart_updated() {
 
     private function base_event() {
         WC()->cart->calculate_totals();
-        
+
         $event = new Drip_Woocommerce_Cart_Event();
         $event->event_action = self::CART_UPDATED_ACTION; // TODO: we can trap cart created events if we have to generate a new cart session id
-        $event->customer_email = wp_get_current_user()->user_email;
+        if ( ! $this->user_invalid() ) {
```

Comment:
Maybe flip the logic to avoid the not?
```
        if ( $this->user_invalid() ) {
            $event->visitor_uuid = $this->find_drip_visitor_uuid();
        } else {
            $event->customer_email = wp_get_current_user()->user_email;
        }
```